feature/split/src/main/java/com/blitzsplit/split/presentation/viewmodel/SplitViewModel.kt

```diff
@@ -0,0 +1,28 @@
+package com.blitzsplit.split.presentation.viewmodel
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+@HiltViewModel
+class SplitViewModel @Inject constructor() : ViewModel() {
+    private val _state: MutableStateFlow<SplitState> = MutableStateFlow(
+        SplitState(billValueTextFieldValue = "", billTitleTextFieldValue = "")
+    )
+    val state: StateFlow<SplitState> = _state
+
+    fun onEvent(event: SplitEvent) {
+        when (event) {
+            is SplitEvent.ChangeBillTitleTextFieldValue -> {
+                _state.update { it.copy(billTitleTextFieldValue = event.value) }
+            }
+
+            is SplitEvent.ChangeBillValueTextFieldValue -> {
+                _state.update { it.copy(billValueTextFieldValue = event.value) }
+            }
```

Comment:
Apenas uma sugestão:

```suggestion
            is SplitEvent.ChangeBillTitleTextFieldValue -> _state.update { 
                it.copy(billTitleTextFieldValue = event.value)
            }

            is SplitEvent.ChangeBillValueTextFieldValue -> _state.update { 
                it.copy(billValueTextFieldValue = event.value)
            }
```